arbitrator/include/capabilities_interface.tpp

```diff
@@ -23,38 +23,91 @@
 #include <functional>
 #include <carma_planning_msgs/srv/plan_maneuvers.hpp>
 #include <rclcpp/exceptions/exceptions.hpp>
-
+#include <chrono>
+#include <thread>
 namespace arbitrator
 {
+    constexpr auto MAX_RETRY_ATTEMPTS {10};
+
     template<typename MSrvReq, typename MSrvRes>
     std::map<std::string, std::shared_ptr<MSrvRes>>
         CapabilitiesInterface::multiplex_service_call_for_capability(const std::string& query_string, std::shared_ptr<MSrvReq> msg)
     {
         std::map<std::string, std::shared_ptr<MSrvRes>> responses;
-
-        for (const auto & topic : get_topics_for_capability(query_string))
+        bool topics_call_successful = false;
+        std::vector<std::string> topics_to_retry;
+        std::vector<std::string> current_topics_to_check;
+        int retry_attempt = 0;
+        while (!topics_call_successful && retry_attempt < MAX_RETRY_ATTEMPTS)
         {
             try {
-                using std::literals::chrono_literals::operator""ms;
+            topics_to_retry = get_topics_for_capability(query_string);
+            topics_call_successful = true;
+            if (retry_attempt > 0)
+            {
+                RCLCPP_INFO_STREAM(rclcpp::get_logger("arbitrator")," Service call to get available strategic plugins successfully finished after retrying: " << retry_attempt);
+            }
+            } catch (const std::exception& error) {
```

Comment:
If there is a specific ROS error we expect to get thrown, we should catch that instead of `std::exception`. I'm assuming it's `rclcpp::exceptions::RCLError` that gets thrown.

Reply:
catching the correct one now